backend/api/invoices/fetch.py

```diff
@@ -48,12 +51,25 @@ def fetch_all_invoices(request: HttpRequest):
     # Validate and sanitize the sort_by parameter
     all_sort_options = ["date_due", "id", "payment_status"]
     context["all_sort_options"] = all_sort_options
-    if sort_by == "date_due":
+
+    if sort_by == "date_due" and default_sort_by["date_due"]:
```

Comment:
Your solution will stop working if you click on the sorting indicator multiple times.
Please test it before pushing.